x-pack/plugins/ingest_manager/public/applications/ingest_manager/index.tsx

```diff
@@ -39,86 +41,175 @@ export const ProtectedRoute: React.FunctionComponent<ProtectedRouteProps> = ({
   return isAllowed ? <Route {...routeProps} /> : <Redirect to={{ pathname: restrictedPath }} />;
 };
 
+const Panel = styled(EuiPanel)`
+  max-width: 500px;
+  margin-right: auto;
+  margin-left: auto;
+`;
+
+const ErrorLayout = ({ children }: { children: JSX.Element }) => (
+  <EuiErrorBoundary>
+    <DefaultLayout>
+      <WithoutHeaderLayout>{children}</WithoutHeaderLayout>
+    </DefaultLayout>
+  </EuiErrorBoundary>
+);
+
 const IngestManagerRoutes = ({ ...rest }) => {
   const { epm, fleet } = useConfig();
+  const { notifications } = useCore();
 
+  const [isPermissionsLoading, setIsPermissionsLoading] = useState<boolean>(false);
+  const [permissionsError, setPermissionsError] = useState<string>();
   const [isInitialized, setIsInitialized] = useState(false);
   const [initializationError, setInitializationError] = useState<Error | null>(null);
 
   useEffect(() => {
     (async () => {
+      setIsPermissionsLoading(false);
+      setPermissionsError(undefined);
       setIsInitialized(false);
       setInitializationError(null);
       try {
-        const res = await sendSetup();
-        if (res.error) {
-          setInitializationError(res.error);
+        setIsPermissionsLoading(true);
+        const permissionsResponse = await sendGetPermissionsCheck();
+        setIsPermissionsLoading(false);
+        if (permissionsResponse.data?.success) {
+          try {
```

Comment:
nitpick: maybe put this in a new function for readability, the level of indentation start to be hard to read